### PR TITLE
Implement threaded search in rest of modes(except empty folders)

### PR DESCRIPTION
**Pull request by qarmin**
Probably something more generic should be used, but for now it is just OK

### CODE REVIEW

File: czkawka_core/src/duplicate.rs [hunk 1]
```diff
@@ -556,130 +522,143 @@ impl DuplicateFinder {
...
-                        self.files_with_identical_size.entry(metadata.len()).or_insert_with(Vec::new);
-                        self.files_with_identical_size.get_mut(&metadata.len()).unwrap().push(fe);
-                    }
+            // Process collected data
+            for (segment, warnings, fe_result) in segments {
+                folders_to_check.extend(segment);
+                self.text_messages.warnings.extend(warnings);
+                for (length, fe) in fe_result {
+                    self.files_with_identical_size.entry(length).or_insert_with(Vec::new);
+                    self.files_with_identical_size.get_mut(&length).unwrap().push(fe);
```
Comment (pczarn) [inline czkawka_core/src/duplicate.rs]: This may be, with additional changes in loop above:
```
for fe in fe_result {
    self.files_with_identical_size.entry(fe.size).or_insert_with(Vec::new);
    self.files_with_identical_size.get_mut(&fe.size).unwrap().push(fe);
```

File: czkawka_core/src/duplicate.rs [hunk 2]
```diff
@@ -44,33 +44,6 @@ pub enum CheckingMethod {
...
-
-impl MyHasher for xxhash_rust::xxh3::Xxh3 {
-    fn update(&mut self, bytes: &[u8]) {
-        self.write(bytes);
-    }
-    fn finalize(&self) -> String {
-        self.finish().to_string()
-    }
-}
-
```
Comment (pczarn) [inline czkawka_core/src/duplicate.rs:73]: why moving this to the end of file?

File: czkawka_core/src/empty_files.rs [hunk 3]
```diff
@@ -166,99 +167,116 @@ impl EmptyFiles {
...
+                            }
+
+                            dir_result.push(next_folder);
+                        } else if metadata.is_file() {
+                            atomic_file_counter.fetch_add(1, Ordering::Relaxed);
+
+                            let file_name_lowercase: String = match entry_data.file_name().into_string() {
+                                Ok(t) => t,
+                                Err(_inspected) => {
+                                    println!("File {:?} has not valid UTF-8 name", entry_data);
```
Comment (pczarn) [inline czkawka_core/src/empty_files.rs]: why would this be a `println!` and not a warning?

File: czkawka_core/src/same_music.rs [hunk 4]
```diff
@@ -229,97 +235,133 @@ impl SameMusic {
...
 
-                        let allowed_extensions = [".mp3", ".flac", ".m4a"];
+                            if !self.allowed_extensions.matches_filename(&file_name_lowercase) {
+                                continue 'dir;
+                            }
 
-                        if !allowed_extensions.iter().any(|r| current_file_name.to_string_lossy().ends_with(r)) {
-                            continue 'dir;
-                        }
+                            if ![".mp3", ".flac", ".m4a"].iter().any(|r| file_name_lowercase.ends_with(r)) {
```
Comment (pczarn) [inline czkawka_core/src/same_music.rs]: could you implement an intersection of `allowed_extensions` set with set `[".mp3", ".flac", ".m4a"]`? Thus replacing two checks with a check against set intersection.

File: czkawka_core/src/similar_images.rs [hunk 5]
```diff
@@ -280,100 +286,124 @@ impl SimilarImages {
...
-                                    self.text_messages.warnings.push(format!("Unable to get modification date from file {}, reason {}", current_file_name.display(), e));
-                                    0
-                                } // Permissions Denied
-                            },
-
-                            hash: Vec::new(),
-                            similarity: Similarity::None,
-                        };
+                            let allowed_image_extensions = [".jpg", ".jpeg", ".png" /*, ".bmp"*/, ".tiff", ".tif", ".tga", ".ff" /*, ".gif"*/, ".jif", ".jfi" /*, ".webp"*/]; // webp cannot be seen in preview, gif needs to be enabled after releasing image crate 0.24.0, bmp needs to be fixed in image crate
+                            if !allowed_image_extensions.iter().any(|e| file_name_lowercase.ends_with(e)) {
```
Comment (pczarn) [inline czkawka_core/src/similar_images.rs]: same concern about set intersection as before

File: czkawka_core/src/similar_videos.rs [hunk 6]
```diff
@@ -243,106 +243,124 @@ impl SimilarVideos {
...
+                                    continue 'dir;
+                                }
+                            }
+                            .to_lowercase();
+
+                            if !self.allowed_extensions.matches_filename(&file_name_lowercase) {
+                                continue 'dir;
+                            }
+
+                            if ![".mp4", ".mpv", ".flv", ".mp4a", ".webm", ".mpg", ".mp2", ".mpeg", ".m4p", ".m4v", ".avi", ".wmv", ".qt", ".mov", ".swf", ".mkv"]
```
Comment (pczarn) [inline czkawka_core/src/similar_videos.rs]: same concern with set intersection

Comment (pczarn) [reply czkawka_core/src/similar_videos.rs]: also: such a big constant should be moved outside of the function into a `const`

File: czkawka_core/src/similar_videos.rs [hunk 7]
```diff
@@ -243,106 +243,124 @@ impl SimilarVideos {
...
+                                        Err(e) => {
+                                            warnings.push(format!("Unable to get modification date from file {}, reason {}", current_file_name.display(), e));
+                                            0
+                                        }
+                                    },
+                                    vhash: Default::default(),
+                                    error: "".to_string(),
+                                };
+
+                                fe_result.push((entry_data.file_name().to_string_lossy().to_string(), fe));
```
Comment (pczarn) [inline czkawka_core/src/similar_videos.rs]: was full path with `current_folder.join(entry_data.file_name())`, now is `entry_data.file_name()`, why the change and what does that mean?

**Review by pczarn — APPROVED** [review]

**Review by pczarn — APPROVED** [review]

**Comment by pczarn** [comment]
Looks good to me.

File: czkawka_core/src/duplicate.rs (hunk 2 above)
Comment (qarmin) [reply czkawka_core/src/duplicate.rs:73]: Without big reasons,
Just I like to have the most important and basic things at the top of file like e.g. enums
